Client/src/main/java/org/projectcardboard/client/models/gui/DeckBox.java

```diff
@@ -178,4 +187,23 @@ private Font getFont(String name) {
     double size = Math.min(45 * SCALE, 45 * SCALE * 8 / name.length());
     return Font.font("DejaVu Sans Light", FontWeight.EXTRA_LIGHT, size);
   }
+
+  private String getFactionCardCount(Deck deck) {
+    // Returns the count of all faction cards in deck with faction names shortened to two chars.
+    // For example a deck with 5 songhai cards and 35 mercs would return "So: 5 Me: 35"
+
+    HashMap<String, Integer> factionCount = new HashMap<>();
+
+    for (Card card : deck.getCards()) {
+      factionCount.merge(card.getFaction(), 1, Integer::sum);
+    }
+
+    ArrayList<String> factions = new ArrayList<>();
+
+    // Shorten faction name to 2 chars; "vanar" -> "Va"
+    factionCount.forEach((key, value) -> factions
+        .add(key.substring(0, 1).toUpperCase() + key.substring(1, 2) + " : " + value.toString()));
+
+    return String.join("  ", factions);
+  }
```

Comment:
```suggestion
/**
   * Returns the count of all faction cards in deck with faction names shortened to two chars. For
   * example a deck with 5 songhai cards and 35 mercs would return "So: 5 Me: 35"
   */
  private String getFactionCardCount(Deck deck) {
    Map<String, Long> factionToCount =
        deck.getCards().stream().collect(Collectors.groupingBy(card -> {
          String faction = card.getFaction();
          return faction.substring(0, 1).toUpperCase() + faction.substring(1, 2);
        }, Collectors.counting()));

    return factionToCount.entrySet().stream()
        .map(entry -> entry.getKey() + " : " + entry.getValue().toString())
        .collect(Collectors.joining("  "));
  }
```

Made it a Javadocs comment and wrote it using streams/the functional API to avoid mutating directly. Pretty much does the exact same thing though, so it's of questionable value. When I started I was hoping there was support for Tuples or Pairs, because it'd make it a bit simpler, wouldn't need to go to a `Map`.